File: 904. Fruit Into Baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        count = Counter()
        i, j, ans, n = 0, 0, 0, len(fruits)
        while j < n:
            count[fruits[j]] += 1
            while len(count) > 2:
                count[fruits[i]] -= 1
                if not count[fruits[i]]:
                    count.pop(fruits[i])
                i += 1
            j += 1
            ans = max(ans, j - i)
        return ans

File: Sliding Window/904. Fruit Into Baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) 
    {
        unordered_map<int, int>mp;
        int ans = 0, i = 0, j = 0, n = fruits.size();
        while(j < n)
        {
            mp[fruits[j++]]++;
            while(mp.size() > 2)
            {
                mp[fruits[i]]--;
                if(mp[fruits[i]] == 0)
                    mp.erase(fruits[i]);
                i++;
            }
            ans = max(ans, j - i);
        }
        return ans;
    }
};